### ISSUE TITLE
Model evaluation

### ISSUE THREAD

**Issue by bllfpc** [bllfpc/TTokDak#6]
학습이 잘 되지 않음. image가 크지않아서 일 수도 있다는 생각이 들긴함.

처음 learning rate를 adam(lr=0.00001)을 두고 돌렸는데 너무 학습이 느린가라고 생각돼서 30 epochs만 확인하고 10배 키운 후 다기 500 epochs 확인. 그러나 거의 학습되지않음

**Comment by bllfpc** [comment]
추가로 weighted_label_dice_coefficient_loss를 주었음

**Comment by bllfpc** [comment]
CT가 MR보다 더 심장이 많이 보여져서 그런지 학습속도는 확실히 더 빠르다.

- MR
Epoch 19/500 - 14s
 - loss: -9.9746e-03 - dice_coefficient: 0.0163
 - label_0_DSC: 0.0161 - label_1_DSC: 0.0312 - label_2_DSC: 0.0325 - label_3_DSC: 6.1031e-05 - label_4_DSC: 6.1031e-05 - label_5_DSC: 6.1031e-05 - label_6_DSC: 6.1031e-05 - label_7_DSC: 6.1031e-05 
- val_loss: -1.1937e-02 - val_dice_coefficient: 0.0169 
- val_label_0_DSC: 0.0164 - val_label_1_DSC: 0.0498 - val_label_2_DSC: 0.0293 - val_label_3_DSC: 6.1031e-05 - val_label_4_DSC: 6.1031e-05 - val_label_5_DSC: 6.1031e-05 - val_label_6_DSC: 6.1031e-05 - val_label_7_DSC: 6.1031e-05

- CT
Epoch 19/30 - 14s 
 - loss: -1.1292e-02 - dice_coefficient: 0.0167
 - label_0_DSC: 0.0162 - label_1_DSC: 0.0402 - label_2_DSC: 0.0340 - label_3_DSC: 6.1031e-05 - label_4_DSC: 6.1031e-05 - label_5_DSC: 6.1031e-05 - label_6_DSC: 6.1031e-05 - label_7_DSC: 6.1031e-05 
- val_loss: -1.3186e-02 - val_dice_coefficient: 0.0175 
- val_label_0_DSC: 0.0165 - val_label_1_DSC: 0.0529 - val_label_2_DSC: 0.0361 - val_label_3_DSC: 6.1031e-05 - val_label_4_DSC: 6.1031e-05 - val_label_5_DSC: 6.1031e-05 - val_label_6_DSC: 6.1031e-05 - val_label_7_DSC: 6.1031e-05

**Comment by bllfpc** [comment]
mask label 값이 전처리에서 잘못돼서 학습이 안됐음

**Comment by bllfpc** [comment]
전처리를 제대로 진행하고 해도 학습이 잘 되지 않는다. 전체 class에 대해서 학습이 되지만 DSC가 10%를 넘는 class가 없다.

그래서 2D로 이미지를 바꿔서 진행하고 있다.

**Comment by bllfpc** [comment]
2D image가 성능이 잘 나오지 않아서 binary로 background와 아닌것을 기준으로 학습했더니 엄청 잘나왔다. 

학습 속도 면에서도 2D U net은 epoch당 160초, 3D U net은 epoch당 30초 정도의 시간이 걸렸다.

한가지 3D U net에서 참고해햐할 부분은 10epoch 정도에서는 크게 학습이 잘 되지 않다가 갑자기 70%로 학습이 확 올랐다.

**Comment by bllfpc** [comment]
Fine tuning을 해봐야할것같다.
